### PR TITLE
NEXUS-8481 Nuget Hosted ITs

### PR DESCRIPTION
**Pull request by mrprescott**
https://issues.sonatype.org/browse/NEXUS-8481

### CODE REVIEW

File: components/nexus-repository/src/main/java/org/sonatype/nexus/repository/storage/AssetEntityAdapter.java [hunk 1]
```diff
@@ -136,8 +136,8 @@ protected void writeFields(final ODocument document, final Asset entity) {
   }
 
   Iterable<Asset> browseByComponent(final ODatabaseDocumentTx db, final Component component) {
-    checkNotNull(component);
-    checkState(component.isPersisted());
+    checkNotNull("component", component);
```
Comment (kellyrob99) [inline components/nexus-repository/src/main/java/org/sonatype/nexus/repository/storage/AssetEntityAdapter.java]: args are backwards here?

Comment (mrprescott) [inline components/nexus-repository/src/main/java/org/sonatype/nexus/repository/storage/AssetEntityAdapter.java]: ack

File: plugins/basic/nexus-repository-nuget/src/test/java/com/sonatype/nexus/repository/nuget/odata/ODataUtilsTest.java [hunk 2]
```diff
@@ -37,4 +42,22 @@ public void testQuery() throws Exception {
...
+    Map<String, String> query = Maps.newHashMap();
+    query.put("$filter", "IsLatestVersion");
+
+    final ComponentQuery foo = ODataUtils.query(query, false);
+
+    assertThat(foo.getWhere(), is("((attributes.nuget.is_latest_version = true))"));
+  }
+
+  @Test
+  public void testBackReference() {
```
Comment (kellyrob99) [inline plugins/basic/nexus-repository-nuget/src/test/java/com/sonatype/nexus/repository/nuget/odata/ODataUtilsTest.java]: incomplete or unneeded?

File: testsuite/modern-testsuite/pom.xml [hunk 3]
```diff
@@ -63,6 +63,19 @@
...
     </dependency>
+
+    <dependency>
+      <groupId>org.sonatype.nexus.plugins</groupId>
+      <artifactId>nexus-repository-nuget</artifactId>
+      <scope>test</scope>
+    </dependency>
+
+    <dependency>
+      <groupId>org.apache.httpcomponents
```
Comment (kellyrob99) [inline testsuite/modern-testsuite/pom.xml]: bad formatting here

File: plugins/basic/nexus-repository-nuget/src/test/java/com/sonatype/nexus/repository/nuget/odata/ODataUtilsTest.java (hunk 2 above)
Comment (mrprescott) [inline plugins/basic/nexus-repository-nuget/src/test/java/com/sonatype/nexus/repository/nuget/odata/ODataUtilsTest.java]: unneeded

File: testsuite/modern-testsuite/pom.xml (hunk 3 above)
Comment (mrprescott) [inline testsuite/modern-testsuite/pom.xml]: thx

File: testsuite/modern-testsuite/src/test/java/org/sonatype/nexus/testsuite/nuget/NugetHostedIT.java [hunk 4]
```diff
@@ -0,0 +1,133 @@
...
+  @After
+  public void deleteHostedRepository() throws Exception {
+    deleteRepository(HOSTED_REPO);
+  }
+
+  @Test
+  public void hostedRepositoryIsAvailable() throws Exception {
+    final String repositoryMetadata = nuget.getRepositoryMetadata();
+    assertThat(repositoryMetadata, is(notNullValue()));
+    assertThat(repositoryMetadata.contains("<Schema Namespace=\"NuGetGallery\""), is(true));
```
Comment (kellyrob99) [inline testsuite/modern-testsuite/src/test/java/org/sonatype/nexus/testsuite/nuget/NugetHostedIT.java]: Usually better expressed using an explicit string matcher, leads to a more meaningful message on error:
```
assertThat(repositoryMetadata, containsString("<Schema Namespace=\"NuGetGallery\""));
```

File: testsuite/modern-testsuite/src/test/java/org/sonatype/nexus/testsuite/nuget/NugetHostedIT.java [hunk 5]
```diff
@@ -0,0 +1,133 @@
...
+    String feed = nuget.feedXml(
+        "Search()?$filter=IsLatestVersion&$orderby=DownloadCount%20desc,Id&$skip=0&$top=30&searchTerm=''&targetFramework='net45'&includePrerelease=false");
+
+    final List<Map<String, String>> entries = parseFeedXml(feed);
+
+    assertThat("entry count", entries.size(), is(1));
+    assertThat("entry ID", entries.get(0).get("ID"), is("SONATYPE.TEST"));
+
+    final int delete = nuget.delete("SONATYPE.TEST", "1.0");
+    assertThat(delete, is(HttpStatus.NO_CONTENT));
```
Comment (kellyrob99) [inline testsuite/modern-testsuite/src/test/java/org/sonatype/nexus/testsuite/nuget/NugetHostedIT.java]: to be complete we should execute a count or feed query and ensure it's gone

File: testsuite/modern-testsuite/src/test/java/org/sonatype/nexus/testsuite/nuget/NugetITSupport.java [hunk 6]
```diff
@@ -0,0 +1,110 @@
...
+    config.attributes("storage").set("writePolicy", WritePolicy.ALLOW.toString());
+    return config;
+  }
+
+  /**
+   * Creates a repository, first removing an existing one if necessary.
+   */
+  protected Repository createOrReplaceRepository(final Configuration config) throws Exception {
+    waitFor(responseFrom(nexusUrl));
+    repositoryManager.get(config.getRepositoryName());
```
Comment (kellyrob99) [inline testsuite/modern-testsuite/src/test/java/org/sonatype/nexus/testsuite/nuget/NugetITSupport.java]: ?

File: testsuite/modern-testsuite/src/test/java/org/sonatype/nexus/testsuite/nuget/NugetITSupport.java [hunk 7]
```diff
@@ -0,0 +1,110 @@
...
+    @Override
+    public void consume(final Map<String, String> data) {
+      entries.add(data);
+    }
+
+    public List<Map<String, String>> getEntries() {
+      return entries;
+    }
+  }
+  //protected NugetClient nugetClient(final Repository repository) throws Exception {
```
Comment (kellyrob99) [inline testsuite/modern-testsuite/src/test/java/org/sonatype/nexus/testsuite/nuget/NugetITSupport.java]: cruft

**Comment by kellyrob99** [comment]
still a bit of cleanup needed but +1 otherwise

File: testsuite/modern-testsuite/src/test/java/org/sonatype/nexus/testsuite/nuget/NugetHostedIT.java (hunk 4 above)
Comment (mrprescott) [inline testsuite/modern-testsuite/src/test/java/org/sonatype/nexus/testsuite/nuget/NugetHostedIT.java]: good idea

File: testsuite/modern-testsuite/src/test/java/org/sonatype/nexus/testsuite/nuget/NugetHostedIT.java (hunk 5 above)
Comment (mrprescott) [inline testsuite/modern-testsuite/src/test/java/org/sonatype/nexus/testsuite/nuget/NugetHostedIT.java]: ack

File: testsuite/modern-testsuite/src/test/java/org/sonatype/nexus/testsuite/nuget/NugetITSupport.java (hunk 6 above)
Comment (mrprescott) [inline testsuite/modern-testsuite/src/test/java/org/sonatype/nexus/testsuite/nuget/NugetITSupport.java]: thanks, guts missing obviously

**Comment by adreghiciu** [comment]
I would add javadoc in from of test methods explaining the scope of that test.
+1

**Comment by mrprescott** [comment]
http://bamboo.s/browse/NX3-OSSF257-3 :white_check_mark:

**Comment by mrprescott** [comment]
Manually merged a85f06b

**Pull request closed without merging**